cfme/cloud/tenant.py

```diff
@@ -67,6 +67,24 @@ def wait_for_disappear(self, timeout=300):
         except TimedOutError:
             logger.error('Timed out waiting for tenant to disappear, continuing')
 
+    def wait_for(self, timeout=300):
+        try:
+            return wait_for(self.exists,
+                            fail_condition=False,
+                            timeout=timeout,
+                            message='Wait for cloud tenant to appear',
+                            delay=10,
+                            fail_func=sel.refresh)
+        except TimedOutError:
+            logger.error('Timed out waiting for tenant to appear, continuing')
+
+    def update(self, updates, wait=True):
+        navigate_to(self, 'Edit')
+        updated_name = updates.get('name', self.name + '_edited')
+        create_tenant_form.fill({'name': updated_name})
+        sel.click(create_tenant_form.save_button)
+        self.wait_for(600)
```

Comment:
**REQUIRED:** So, the comment previously said, there is no need for a wait_for method, and to put the actual wait_for at the end of the update method. 
**REQUIRED:** The method also above is bad as it just silently passes if the provider doesn't appear. This is going to cause you bad debugging in the future as you will not see the error when it happens.